### PR TITLE
[Metrics SDK] Cleanup ENABLE_METRICS_PREVIEW

### PR DESCRIPTION
**Pull request by esigo**
## Changes

This does cleanup as part of #1573 - remove ENABLE_METRICS_PREVIEW macro check in new metrics implementation.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

File: sdk/CMakeLists.txt [hunk 1]
```diff
@@ -25,9 +25,7 @@ if(NOT WITH_LOGS_PREVIEW)
 endif()
 
 set(METRICS_EXCLUDE_PATTERN "")
-if(NOT WITH_METRICS_PREVIEW)
-  set(METRICS_EXCLUDE_PATTERN "_metrics")
-endif()
+set(METRICS_EXCLUDE_PATTERN "_metrics")
```
Comment (marcalff) [inline sdk/CMakeLists.txt:28]: Once `_metrics` are removed (PR #1734), `METRICS_EXCLUDE_PATTERN` can probably be removed as well.

**Review by marcalff — APPROVED** [review]
LGTM

File: api/CMakeLists.txt [hunk 2]
```diff
@@ -19,9 +19,7 @@ if(NOT WITH_LOGS_PREVIEW)
 endif()
 
 set(METRICS_EXCLUDE_PATTERN "")
-if(NOT WITH_METRICS_PREVIEW)
-  set(METRICS_EXCLUDE_PATTERN "metrics")
-endif()
+set(METRICS_EXCLUDE_PATTERN "metrics")
```
Comment (lalitb) [inline api/CMakeLists.txt:22]: we don't need this variable anymore.

**Review by lalitb — APPROVED** [review]
LGTM. Thanks for the cleanup :)

File: sdk/CMakeLists.txt (hunk 1 above)
Comment (esigo) [reply sdk/CMakeLists.txt]: thanks, removed

File: api/CMakeLists.txt (hunk 2 above)
Comment (esigo) [reply api/CMakeLists.txt]: thanks, cleaned

**Pull request merged**